content/en/blog/2023/dlb-connection-balancing/index.md

```diff
@@ -0,0 +1,307 @@
+---
+title: "Using Accelerated Offload Connection Load Balancing in Istio"
+description: "Accelerate connection balancing using DLB connection balancing configuration in Istio gateways."
+publishdate: 2023-06-08
+attribution: "Loong Dai (Intel)"
+keywords: [Istio, DLB, gateways]
+---
+
+## What is connection load balancing?
+
+Load balancing is a core networking solution used to distribute traffic across multiple servers in a server farm.
+Load balancers improve application availability and responsiveness and prevent server overload. Each load balancer
+sits between client devices and backend servers, receiving and then distributing incoming requests to any available server capable of fulfilling them.
+
+For a common web server, it usually has multiple workers (processors or threads). If many clients connect to
+a single worker, this worker becomes busy and brings long tail latency while other workers run in the free state,
+affecting the performance of the web server. Connection load balancing is the solution for this situation,
+which is also known as connection balancing.
+
+## What does Istio do for connection load balancing?
+
+Istio uses Envoy as the data plane.
+
+Envoy provides a connection load balancing implementation called Exact connection balancer. As its name says, a lock is held during balancing so that connection counts are nearly exactly balanced between workers. It is "nearly" exact in the sense that a connection might close in parallel thus making the counts incorrect, but this should be rectified on the next accept. This balancer sacrifices accept throughput for accuracy and should be used when there are a small number of connections that rarely cycle, e.g., service mesh gRPC egress.
+
+Obviously, it is not suitable for an ingress gateway since an ingress gateway accepts thousands of connections within a short time, and the resource cost from the lock brings a big drop in throughput.
+
+Now, Envoy has an integrated Intel® Dynamic Load Balancing (Intel®DLB) connection load balancing to accelerate in big connection cases like ingress gateway.
+
+## How Intel® Dynamic Load Balancing accelerates connection load balancing in Envoy
+
+Intel DLB is a hardware managed system of queues and arbiters connecting producers and consumers. It is a PCI device envisaged to live in the server CPU uncore and can interact with software running on cores, and potentially with other devices.
```

Comment:
I didn't know what an uncore is, so you might want to link https://en.wikipedia.org/wiki/Uncore or similar, or add to the glossary.

Reply:
Added.